service/src/main/java/bio/terra/landingzone/stairway/flight/create/resource/step/CreateLogAnalyticsWorkspaceStep.java

```diff
@@ -0,0 +1,132 @@
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext context) throws InterruptedException {
+    String logAnalyticsWorkspaceId =
+        context.getWorkingMap().get(LOG_ANALYTICS_WORKSPACE_ID, String.class);
+    try {
+      if (logAnalyticsWorkspaceId != null) {
+        armManagers.logAnalyticsManager().workspaces().deleteById(logAnalyticsWorkspaceId);
+        logger.info("{} resource with id={} deleted.", getResourceType(), logAnalyticsWorkspaceId);
+      }
+      // Deploying AKS with monitoring connected to a log analytics workspace also deploys a
+      // container insights solution named `ContainerInsights(WORKSPACE_ID)` which is untagged
+      // this code lists them all and later code figures out which to delete
+      var solutions =
+          armManagers
+              .azureResourceManager()
+              .genericResources()
+              .listByResourceGroup(getMRGName(context))
+              .stream()
+              .filter(
+                  r ->
+                      AzureResourceTypeUtils.AZURE_SOLUTIONS_TYPE.equalsIgnoreCase(
+                          "%s/%s".formatted(r.resourceProviderNamespace(), r.resourceType())))
+              .toList();
+      solutions.forEach(
+          s -> {
+            armManagers.azureResourceManager().genericResources().deleteById(s.id());
+            logger.info(
+                "{} resource with id={} deleted.",
+                AzureResourceTypeUtils.AZURE_SOLUTIONS_TYPE,
+                s.id());
+          });
+    } catch (ManagementException e) {
+      if (StringUtils.equalsIgnoreCase(e.getValue().getCode(), "ResourceNotFound")) {
+        logger.error(
+            "Log analytics workspace doesn't exist or has been already deleted. Id={}",
+            logAnalyticsWorkspaceId);
+        return StepResult.getStepResultSuccess();
+      }
+      logger.error(
+          "Failed attempt to delete log analytics workspace. Id={}", logAnalyticsWorkspaceId);
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, e);
+    }
+    return StepResult.getStepResultSuccess();
+  }
+
+  @Override
+  protected void createResource(FlightContext context, ArmManagers armManagers) {
+    var landingZoneId =
+        getParameterOrThrow(
+            context.getInputParameters(), LandingZoneFlightMapKeys.LANDING_ZONE_ID, UUID.class);
+    var logAnalyticsName =
+        resourceNameGenerator.nextName(
+            ResourceNameGenerator.MAX_LOG_ANALYTICS_WORKSPACE_NAME_LENGTH);
+
+    var logAnalyticsWorkspace =
+        armManagers
+            .logAnalyticsManager()
+            .workspaces()
+            .define(logAnalyticsName)
+            .withRegion(getMRGRegionName(context))
+            .withExistingResourceGroup(getMRGName(context))
+            .withRetentionInDays(
+                context
+                    .getInputParameters()
+                    .get(
+                        CromwellBaseResourcesFactory.ParametersNames.AUDIT_LOG_RETENTION_DAYS
```

Comment:
Should this come from `CromwellBaseResourcesFactory`? It seems likely that we would want different retention periods for protected data landing zones vs. the base one.